proposals/0133-syscall-get-epoch-stake.md

```diff
@@ -0,0 +1,98 @@
+---
+simd: '0133'
+title: Syscall Get-Epoch-Stake
+authors:
+  - Joe Caulfield (Anza)
+category: Standard
+type: Core
+status: Draft
+created: 2024-03-25
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+A syscall to retrieve a vote account's delegated stake for the current epoch.
+
+## Motivation
+
+Currently, on-chain programs have no knowledge of the current epoch's stake and
+how much active stake is delegated to a certain vote account.
+
+If this data was available for querying by on-chain programs, it would unblock
+many use cases, such as validator governance and secondary consensus mechanisms,
+that were previously not possible on Solana.
+
+Additionally, this would enable the Feature Gate program defined in
+[SIMD 0089](./0089-programify-feature-gate.md) to tally vote account stake in
+support for a pending feature gate.
+
+## Alternatives Considered
+
+An alternative design has been proposed in the past for using an on-chain sysvar
+account to store all of the current epoch's stake. It was proposed in
+[SIMD 0056](https://github.com/solana-foundation/solana-improvement-documents/pull/56).
+
+Using a sysvar account to store even just the current epoch's stake introduces a
+size limitation on the number of entries that can be stored, which the number of
+validators on Solana could surpass in the future.
+
+By offering access to this information through only a syscall, we can avoid this
+account maximum size constraint. However, the syscall approach also makes the
+data more cumbersome to retrieve off-chain than the sysvar approach.
+
+While retrieving this data off-chain may not be straightforward, it remains
+entirely feasible. Conversely, on-chain programs currently do not have this
+capability at all.
```

Comment:
```suggestion
Exposing epoch-stake information through a syscall avoids this
account maximum size constraint. While the syscall approach does not offer the 
easy off-chain retrieval of a sysvar, there are existing ways
to get epoch-stake data off chain. The priority of a new design should be making the data available to on-chain programs.
```
I think this is what you're trying to say.

Reply:
Yeah, exactly, thanks.